src/lib/selections.test.ts

```diff
@@ -0,0 +1,1232 @@
+import { expect } from 'vitest'
+import { assertParse, initPromise, ArtifactGraph } from 'lang/wasm'
+import { getNodePathFromSourceRange } from 'lang/queryAstNodePathUtils'
+import { codeToIdSelections, Selection } from './selections'
+import { buildArtifactIndex } from './artifactIndex'
+
+beforeAll(async () => {
+  await initPromise
+})
+
+describe('testing source range to artifact conversion', () => {
+  const MY_CODE = `sketch001 = startSketchOn('XZ')
+profile001 = startProfileAt([105.55, 105.55], sketch001)
+  |> xLine(332.55, %, $seg01)
+  |> yLine(-310.12, %, $seg02)
+  |> xLine(-373.65, %)
+  |> line(endAbsolute = [profileStartX(%), profileStartY(%)])
+  |> close()
+extrude001 = extrude(profile001, length = 500)
+
+sketch002 = startSketchOn(extrude001, seg01)
+profile002 = startProfileAt([-321.34, 361.76], sketch002)
+  |> line(end = [109.03, -61.79])
+  |> line(end = [-124.48, -132.65])
+  |> line(endAbsolute = [profileStartX(%), profileStartY(%)])
+  |> close()
+extrude002 = extrude(profile002, length = 500)
+sketch005 = startSketchOn(extrude002, 'END')
+profile006 = circle({
+  center = [-292.57, 302.55],
+  radius = 25.89
+}, sketch005)
+sketch004 = startSketchOn(extrude001, seg02)
+profile005 = startProfileAt([36.1, 174.49], sketch004)
+  |> angledLine([0, 22.33], %, $rectangleSegmentA003)
+  |> angledLine([
+       segAng(rectangleSegmentA003) - 90,
+       155.27
+     ], %)
+  |> angledLine([
+       segAng(rectangleSegmentA003),
+       -segLen(rectangleSegmentA003)
+     ], %)
+  |> line(endAbsolute = [profileStartX(%), profileStartY(%)])
+  |> close()
+sketch003 = startSketchOn(extrude001, seg02)
+profile003 = startProfileAt([-115.59, 439.4], sketch003)
+  |> angledLine([0, 130.08], %, $rectangleSegmentA002)
+  |> angledLine([
+       segAng(rectangleSegmentA002) - 90,
+       123.84
+     ], %)
+  |> angledLine([
+       segAng(rectangleSegmentA002),
+       -segLen(rectangleSegmentA002)
+     ], %)
+  |> line(endAbsolute = [profileStartX(%), profileStartY(%)])
+  |> close()
+profile004 = circle({
+  center = [-88.54, 209.41],
+  radius = 42.72
+}, sketch003)
+`
+  const ___artifactGraph = new Map([
```

Comment:
@jtran I ended up pasting the entire artifactGraph for this snippet in, because there's no artifactGraph in mock mode, didn't want to do a real execution, but maybe that's better?

Reply:
I think this is fine for now. Maybe this is another reason to prioritize #5093.